opentelemetry-sdk/src/opentelemetry/sdk/_configuration/__init__.py

```diff
@@ -161,7 +191,8 @@ def _init_tracing(
             ResourceAttributes.TELEMETRY_AUTO_VERSION
         ] = auto_instrumentation_version
     provider = TracerProvider(
-        id_generator=id_generator(),
+        id_generator=id_generator,
```

Comment:
Any idea why this was changed?

Reply:
We were previously passing in a class not an implementation. Both because that was bad and to make it more inline with the other arguments that have defaults, I fixed this in _import_id_generator